src/main/java/com/googlecode/objectify/impl/TransactorNo.java

```diff
@@ -85,27 +85,22 @@ public TransactionImpl getTransaction() {
 		return this.transactNew(parent, Integer.MAX_VALUE, work);
 	}
 
-	/* (non-Javadoc)
+	/*
+	 * It will log out and throw the ConcurrentModificationException so that the caller can manage the datastore "contention" problem on call side.
+	 * I don't want to modify the method declaration, therefore the "limitTries" is in the parameter list but it has no functionality.
 	 * @see com.googlecode.objectify.impl.Transactor#transactNew(com.googlecode.objectify.impl.ObjectifyImpl, int, com.googlecode.objectify.Work)
 	 */
 	@Override
 	public <R> R transactNew(ObjectifyImpl<O> parent, int limitTries, Work<R> work) {
-		Preconditions.checkArgument(limitTries >= 1);
-
-		while (true) {
-			try {
-				return transactOnce(parent, work);
-			} catch (ConcurrentModificationException ex) {
-				if (--limitTries > 0) {
-					if (log.isLoggable(Level.WARNING))
-						log.warning("Optimistic concurrency failure for " + work + " (retrying): " + ex);
-
-					if (log.isLoggable(Level.FINEST))
-						log.log(Level.FINEST, "Details of optimistic concurrency failure", ex);
-				} else {
-					throw ex;
-				}
-			}
+		try {
+			return transactOnce(parent, work);
+		} catch (ConcurrentModificationException ex) {
+			if (log.isLoggable(Level.WARNING))
+				log.warning("Optimistic concurrency failure for " + work + " (retrying): " + ex);

```

Comment:
The log says retrying but it doesn't retry the problem. 
The log messages here should be removed and let the code catching the exception log it. Otherwise we will log twice.

Reply:
https://github.com/aliz-ai/objectify/pull/2

Reply:
and limitTries is completely ignored, this will never be merged into objectify if that is the goal IMO

Reply:
This is a breaking change so I don't expect this to be merged. In v6 this is the default behavior so I guess others run into this problem.